core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractLibrary.java

```diff
@@ -520,4 +520,15 @@ public TagBarcodeFamily getCurrentFamily() {
       return tagBarcodes.get(0).getFamily();
     }
   }
+
+  @Override
+  public SampleTissue getSampleTissue() {
+    if (this.getSample() instanceof SampleAdditionalInfo) {
+
+      for (SampleAdditionalInfo parent = (SampleAdditionalInfo) this.getSample(); parent != null; parent = parent.getParent()) {
+        if (parent instanceof SampleTissue) return (SampleTissue) parent;
```

Comment:
I don't think SampleTissue can parent Libraries; only SampleAnalyteAliquot can do that.